Missing enum in native mode

## description

When calling some git commands in native, an enum is missing: `java.lang.IllegalArgumentException: Enumerated values of type org.eclipse.jgit.transport.HttpConfig$HttpRedirectMode not available`:

```
java.lang.IllegalArgumentException: Enumerated values of type org.eclipse.jgit.transport.HttpConfig$HttpRedirectMode not available
	at org.eclipse.jgit.lib.Config.allValuesOf(Config.java:585)
	at org.eclipse.jgit.lib.Config.getEnum(Config.java:572)
	at org.eclipse.jgit.transport.HttpConfig.init(HttpConfig.java:305)
	at org.eclipse.jgit.transport.HttpConfig.<init>(HttpConfig.java:297)
	at org.eclipse.jgit.transport.TransportHttp.<init>(TransportHttp.java:336)
	at org.eclipse.jgit.transport.TransportHttp$1.open(TransportHttp.java:203)
	at org.eclipse.jgit.transport.Transport.open(Transport.java:558)
	at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:159)
	at org.eclipse.jgit.api.LsRemoteCommand.call(LsRemoteCommand.java:131)
	at eu.pierrebeitz.GreetingCommand.run(GreetingCommand.java:20)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
	at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:40)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:141)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base@21.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: java.lang.NoSuchMethodException: org.eclipse.jgit.transport.HttpConfig$HttpRedirectMode.values()
	at java.base@21.0.2/java.lang.Class.checkMethod(DynamicHub.java:1075)
	at java.base@21.0.2/java.lang.Class.getMethod(DynamicHub.java:1060)
	at org.eclipse.jgit.lib.Config.allValuesOf(Config.java:580)
	... 24 more
```

This is on a Apple M4 Pro running Sequoia 15.5.

## Reproduction

For convenience, a repro is available at https://github.com/PierreBtz/jgit-quarkus-bug, all the steps are in the README.

It looks like a regression since the same code works properly when using [quarkus-jgit](https://github.com/quarkiverse/quarkus-jgit) up to 3.3.3 (first failing version is 3.4.0).

Important note, for a reason I can't explain, I cannot reproduce on the machine compiling the binary, I have to copy it to another osx machine.

## Environment

I reproduced this with:

for the machine compiling the binary:

```
Darwin pbeitz-MacBook-Pro 24.5.0 Darwin Kernel Version 24.5.0: Tue Apr 22 19:53:27 PDT 2025; root:xnu-11417.121.6~2/RELEASE_ARM64_T6041 arm64
```

for the machine running it:

```
Darwin 26853.local 22.6.0 Darwin Kernel Version 22.6.0: Fri Sep 15 13:39:54 PDT 2023; root:xnu-8796.141.3.700.8~1/RELEASE_ARM64_T8112 arm64
```

(not stating the OS version is the issue, it's just what I had at hand to work on the reproducer)

Compiled with `OpenJDK Runtime Environment GraalVM CE 21.0.2+13.1 (build 21.0.2+13-jvmci-23.1-b30)`.

--- a/deployment/src/main/java/io/quarkus/jgit/deployment/JGitProcessor.java
+++ b/deployment/src/main/java/io/quarkus/jgit/deployment/JGitProcessor.java
@@ -14,6 +14,7 @@ import org.eclipse.jgit.lib.Ref;
 import org.eclipse.jgit.lib.RefUpdate;
 import org.eclipse.jgit.lib.SignatureVerifier;
 import org.eclipse.jgit.lib.SubmoduleConfig;
+import org.eclipse.jgit.transport.HttpConfig;
 
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
@@ -54,6 +55,7 @@ class JGitProcessor {
                 DiffAlgorithm.SupportedAlgorithm.class,
                 DirCache.DirCacheVersion.class,
                 GpgConfig.GpgFormat.class,
+                HttpConfig.HttpRedirectMode.class,
                 IndexDiff.StageState.class,
                 JGitText.class,
                 ObjectChecker.ErrorType.class,

